src/signed-http-client.js

```diff
@@ -3,6 +3,11 @@ const aws4 = require('aws4');
 
 let baseurl = process.env.ApiBaseUrl;
 const region = process.env.AWS_REGION || 'eu-west-1';
+let globalHeaders = {};
```

Comment:
Does it really make sense to have the headers as global? E.g. correlation id might not be the same between a POST and a GET performed by the same loaded module, so maybe it makes sense to apply it to the individual methods as an argument instead? Could also do a breaking change and convert the arguments to objects and support destructuring instead.